webapp/components/test-file-results-time-series.html

```diff
@@ -0,0 +1,157 @@
+found in the LICENSE file.
+-->
+
+<link rel="import" href="../bower_components/polymer/lib/elements/dom-repeat.html">
+<link rel="import" href="runs.html">
+
+<dom-module id="test-file-results-time-series">
+  <template>
+    <style>
+      .browser {
+        display: flex;
+        max-height: 150px;
+        overflow: auto;
+      }
+      .browser-name {
+        writing-mode: vertical-rl;
+        text-orientation: sideways-right;
+        border-right: 1px solid black;
+        padding: 8px 4px;
+      }
+      .browser-result {
+        padding: 8px 0;
+      }
+      .browser-results {
+        display: flex;
+      }
+      .browser-result-status {
+        cursor: default;
+        border: 1px solid #ddd;
+        background-color: #eee;
+        width: 10px;
+        height: 10px;
+      }
+      .browser-result-status.OK, .browser-result-status.PASS {
+        background-color: rgb(90, 242, 113);
+      }
+      .browser-result-status.FAIL {
+        background-color: rgb(242, 90, 90);
+      }
+    </style>
+    <template is="dom-repeat" items="[[results]]" as="browserResults">
+      <div class="browser">
+        <div class="browser-name">[[browserResults.browserName]]</div>
+        <div class="browser-results"></div>
+        <template is="dom-repeat" items="[[browserResults.runResults]]" as="runResult">
+            <div class="browser-result">
+              <!-- Tooltip or something associated with run? -->
+              <template is="dom-repeat" items="[[runResult.results]]" as="result">
+                <div
+                    class$="browser-result-status [[result.status]]"
+                    onmouseenter="[[bindResultHover(runResult.run, result.name)]]">&nbsp;</div>
+              </template>
+            </div>
+        </template>
+      </div>
+    </template>
+
+    <pre id="status">&nbsp;</pre>
+  </template>
+  <script>
+    class TestFileResultsTimeSeries extends window.Polymer.Element {
+      static get is() {
+        return 'test-file-results-time-series';
+      }
+
+      static get properties() {
+        return {
+          path: {
+            type: String,
+          },
+          maxCount: {
+            type: Number,
+            value: 20,
+          },
+          labels: {
+            type: Array,
+            value: [],
+          },
+          results: {
+            type: Array,
+            value: [],
+          },
+        };
+      }
+
+      bindResultHover(run, subTestName) {
+        return () => {
+          let statusElement = this.shadowRoot.querySelector('#status');
+          statusElement.textContent =
+              `${run.browser_name} ${run.browser_version} ${run.os_name} ${run.os_version} @ ${run.time_start} : ${subTestName}`;
+
+        }
+      }
+
+      async connectedCallback() {
+        const resp = await window.fetch(`/api/runs?complete=true&max-count=${this.maxCount}&labels=${this.labels.join(',')}`);
```

Comment:
I guess we're not able to build a full run results URL with just the result of /api/shas... bummer.